src/main/java/uk/gov/companieshouse/api/strikeoffobjections/file/FileTransferApiClient.java

```diff
@@ -0,0 +1,125 @@
+package uk.gov.companieshouse.api.strikeoffobjections.file;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.multipart.MultipartFile;
+import uk.gov.companieshouse.api.strikeoffobjections.common.ApiLogger;
+
+import java.io.IOException;
+
+public class FileTransferApiClient {
+
+    private static final String HEADER_API_KEY = "x-api-key";
+    private static final String UPLOAD = "upload";
+    private static final String CONTENT_DISPOSITION_VALUE = "form-data; name=%s; filename=%s";
+    private static final String NULL_RESPONSE_MESSAGE = "null response from file transfer api url";
+
+    @Autowired
+    private ApiLogger logger;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Value("${FILE_TRANSFER_API_URL}")
+    private String fileTransferApiURL;
+
+    @Value("${FILE_TRANSFER_API_KEY}")
+    private String fileTransferApiKey;
+
+
+    private <T> FileTransferApiClientResponse makeApiCall(FileTransferOperation <T> operation,
+                                                          FileTransferResponseBuilder <T> responseBuilder) {
+        FileTransferApiClientResponse response = new FileTransferApiClientResponse();
+
+        try {
+            T operationResponse = operation.execute();
+            response = responseBuilder.createResponse(operationResponse);
+        } catch (IOException e) {
+            logger.errorContext(e.getMessage(), e);
```

Comment:
As briefly mentioned above, the context argument is meant to be the request id of the request.
I would suggest passing the request id into this method and adding it as the context argument
logger.errorContext(requestId, e.getMessage(), e);

Reply:
Added a request id